Move TH_BLAS_MKL flags into ATen/Config.h

peterbell10 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #69419
* #69041
* #69040
* #69039
* #69038

Differential Revision: [D32872480](https://our.internmc.facebook.com/intern/diff/D32872480)

peterbell10 committed:
Move TH_BLAS_MKL flags into ATen/Config.h

[ghstack-poisoned]

peterbell10 committed:
Update on "Move TH_BLAS_MKL flags into ATen/Config.h"




[ghstack-poisoned]

peterbell10 committed:
Update on "Move TH_BLAS_MKL flags into ATen/Config.h"




[ghstack-poisoned]

ngimel commented:
@ngimel has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32872480).

peterbell10 committed:
Update on "Move TH_BLAS_MKL flags into ATen/Config.h"


Differential Revision: [D32872480](https://our.internmc.facebook.com/intern/diff/D32872480)

[ghstack-poisoned]

ngimel commented:
I cannot land this due to weird internal errors, I'm trying to split it into `AT_MKL_ENABLED()` part that shouldn't require any build changes (#70219) and everything else (including internal build changes due to AT_MKL_SEQUENTIAL)
